library/src/main/java/com/novoda/downloadmanager/LiteDownloadMigrationService.java

```diff
@@ -73,19 +72,12 @@ public void startMigration(MigrationCallback migrationCallback) {
     }
 
     private void createNotificationChannel() {
-        Optional<NotificationChannel> notificationChannel = notificationChannelCreator.createNotificationChannel();
-        String channelName = notificationChannelCreator.getNotificationChannelId();
-
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && notificationChannel.isPresent() && notificationChannelDoesNotExist(channelName)) {
-            notificationManager.createNotificationChannel(notificationChannel.get());
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            NotificationChannel notificationChannel = notificationChannelCreator.createNotificationChannel();
+            notificationManager.createNotificationChannel(notificationChannel);
```

Comment:
We also don't need to check if the channel already exists before creating it. Calling `createNotificationChannel` multiple times with identical `NotificationChannel` arguments still results in only one channel being created.

Reply:
Yeah, in the examples too. Google suggests to create the channel without checking anything in the constructor of the class responsible for notifications.